.github/workflows/check-state-compatibility.yml

```diff
@@ -0,0 +1,111 @@
+name: Check Osmosis mainnet
+
+on:
+  workflow_dispatch:
+  pull_request:
+    branches:
+      - "*"
+
+env:
+  GOLANG_VERSION: 1.18
+  GENESIS_URL: https://github.com/osmosis-labs/networks/raw/main/osmosis-1/genesis.json
+  ADDRBOOK_URL: https://dl2.quicksync.io/json/addrbook.osmosis.json
+  SNAPSHOT_BUCKET: https://snapshots.osmosis.zone
+  RPC_ENDPOINT: https://rpc.osmosis.zone
+  LCD_ENDPOINT: https://lcd.osmosis.zone
+  DELTA_HALT_HEIGHT: 50
+
+jobs:
+  compare_versions:
+    # Compare current mainnet osmosis major version with the major version of github branch
+    # Skip the next job if the current github major is greater than the current mainnet major
+    runs-on: self-hosted
+    outputs:
+      mainnet_major_version: ${{ steps.mainnet_version.outputs.mainnet_major_version }}
+    steps:
+      - name: Get mainnet major version
+        id: mainnet_version
+        run: |
+          # Find current major version via rpc.osmosis.zone/abci_info
+          RPC_ABCI_INFO=$(curl -s --retry 5 --retry-delay 5 --connect-timeout 30 -H "Accept: application/json" ${{ env.RPC_ENDPOINT }}/abci_info)
+          MAINNET_MAJOR_VERSION=$(echo $RPC_ABCI_INFO | dasel --plain -r json  'result.response.version' | cut -f 1 -d '.')
+
+          echo "MAINNET_MAJOR_VERSION=$MAINNET_MAJOR_VERSION" >> $GITHUB_ENV
+          echo "mainnet_major_version=$MAINNET_MAJOR_VERSION" >> $GITHUB_OUTPUT
+
+  check_mainnet:
+    # DO NOT CHANGE THIS: please read the note above
+    needs: compare_versions
+    runs-on: self-hosted
+    steps:
+      - name: 🐍 Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.x
+
+      - name: 🐿 Setup Golang
+        uses: actions/setup-go@v3
+        with:
+          go-version: "^${{ env.GOLANG_VERSION }}"
+
+      - name: Checkout osmosis-installer branch
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Checkout osmosis repo
+        uses: actions/checkout@v3
+        with:
+          repository: osmosis-labs/osmosis
+          path: repo
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: 🔨 Build the osmosisd binary
+        run: |
+          cd repo
+          make build
+          build/osmosisd version
```

Comment:
Not needed. Done by installer

Reply:
if we dont run the osmosisd binary the folder "$HOME/.osmosisd/config" doesnot exist